From 1cebea67483e05969215878cb3a026295a74c232 Mon Sep 17 00:00:00 2001
From: Sandro Santilli <strk@kbt.io>
Date: Thu, 7 Sep 2023 16:57:53 +0200
Subject: [PATCH] Normalize results from ST_DelaunayTriangles

GEOS recent commit changed starting point of polygon rings
ref: https://debbie.postgis.net/view/GEOS/job/GEOS_Master/1793/
---
 regress/core/delaunaytriangles.sql      | 19 +++++++++----------
 regress/core/delaunaytriangles_expected | 18 +++++++++---------
 2 files changed, 18 insertions(+), 19 deletions(-)

diff --git a/regress/core/delaunaytriangles.sql b/regress/core/delaunaytriangles.sql
index 18589e05d1e..efc56102931 100644
--- a/regress/core/delaunaytriangles.sql
+++ b/regress/core/delaunaytriangles.sql
@@ -1,12 +1,11 @@
--- TODO: normalize !
-SELECT 1,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9)'::geometry));
-SELECT 2,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry));
-SELECT 3,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2));
-SELECT 4,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2, 1));
-SELECT 5,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9)'::geometry, 0.0, 2));
-SELECT 6,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 0.0, 2));
-SELECT 7,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2.0, 2));
+SELECT 1,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9)'::geometry)));
+SELECT 2,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry)));
+SELECT 3,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2)));
+SELECT 4,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2, 1)));
+SELECT 5,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9)'::geometry, 0.0, 2)));
+SELECT 6,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 0.0, 2)));
+SELECT 7,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5, 6 0, 7 9, 8 9)'::geometry, 2.0, 2)));
 
 -- 3DZ
-SELECT 10,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5 3, 6 0 2, 7 9 1)'::geometry));
-SELECT 11,  ST_AsText(ST_DelaunayTriangles('MULTIPOINT(5 5 1, 6 0 2, 7 9 3)'::geometry, 0.0, 2));
+SELECT 10,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5 3, 6 0 2, 7 9 1)'::geometry)));
+SELECT 11,  ST_AsText(ST_Normalize(ST_DelaunayTriangles('MULTIPOINT(5 5 1, 6 0 2, 7 9 3)'::geometry, 0.0, 2)));
diff --git a/regress/core/delaunaytriangles_expected b/regress/core/delaunaytriangles_expected
index d3ab102bcd7..5b61b195c8d 100644
--- a/regress/core/delaunaytriangles_expected
+++ b/regress/core/delaunaytriangles_expected
@@ -1,9 +1,9 @@
-1|GEOMETRYCOLLECTION(POLYGON((5 5,6 0,7 9,5 5)))
-2|GEOMETRYCOLLECTION(POLYGON((5 5,6 0,8 9,5 5)),POLYGON((5 5,8 9,7 9,5 5)))
-3|GEOMETRYCOLLECTION(POLYGON((5 5,6 0,7 9,5 5)))
-4|MULTILINESTRING((5 5,7 9),(5 5,6 0),(6 0,7 9))
-5|TIN(((5 5,6 0,7 9,5 5)))
-6|TIN(((5 5,6 0,8 9,5 5)),((5 5,8 9,7 9,5 5)))
-7|TIN(((5 5,6 0,7 9,5 5)))
-10|GEOMETRYCOLLECTION Z (POLYGON Z ((5 5 3,6 0 2,7 9 1,5 5 3)))
-11|TIN Z (((5 5 1,6 0 2,7 9 3,5 5 1)))
+1|GEOMETRYCOLLECTION(POLYGON((5 5,7 9,6 0,5 5)))
+2|GEOMETRYCOLLECTION(POLYGON((5 5,8 9,6 0,5 5)),POLYGON((5 5,7 9,8 9,5 5)))
+3|GEOMETRYCOLLECTION(POLYGON((5 5,7 9,6 0,5 5)))
+4|MULTILINESTRING((6 0,7 9),(5 5,7 9),(5 5,6 0))
+5|GEOMETRYCOLLECTION(POLYGON((5 5,7 9,6 0,5 5)))
+6|GEOMETRYCOLLECTION(POLYGON((5 5,8 9,6 0,5 5)),POLYGON((5 5,7 9,8 9,5 5)))
+7|GEOMETRYCOLLECTION(POLYGON((5 5,7 9,6 0,5 5)))
+10|GEOMETRYCOLLECTION Z (POLYGON Z ((5 5 3,7 9 1,6 0 2,5 5 3)))
+11|GEOMETRYCOLLECTION Z (POLYGON Z ((5 5 1,7 9 3,6 0 2,5 5 1)))
